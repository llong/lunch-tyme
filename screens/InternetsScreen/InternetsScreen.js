import React from 'react';
import { Text, View } from 'react-native';

class InternetsScreen extends React.Component {
  render() {
    return (
      <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
        <Text>Settings!</Text>
      </View>
    );
  }
}

export default InternetsScreen;