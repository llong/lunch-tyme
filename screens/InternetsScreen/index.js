import { createStackNavigator } from 'react-navigation';
import InternetsScreen from './InternetsScreen';

const InternetStack = createStackNavigator(
  {
      internet: InternetsScreen,
  },
  {
      defaultNavigationOptions: {
          headerStyle: {
            backgroundColor: '#43E895',
          },
          headerTintColor: '#fff',
          headerTitleStyle: {
            fontFamily: 'Avenir Next Demi Bold',
            fontSize: 17
          },
      },
  }
);

export default InternetStack;
