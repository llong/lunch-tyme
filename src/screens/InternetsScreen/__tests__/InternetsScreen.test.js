import React from 'react';
import { shallow } from 'enzyme';
import InternetsScreen from '../InternetsScreen';

const setParamsMock = jest.fn();

const props = {
  navigation: {
    setParams: setParamsMock,
  },
};

describe('InternetsScreen', () => {
  const wrapper = shallow(<InternetsScreen {...props} />, { disableLifecycleMethods: true });
  it('should render component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should set navigation btn params when mounted', () => {
    const instance = wrapper.instance();
    instance.webview = { current: { goBack: jest.fn() } };
    instance.componentDidMount();
    expect(setParamsMock).toHaveBeenCalledTimes(1);
  });
});
