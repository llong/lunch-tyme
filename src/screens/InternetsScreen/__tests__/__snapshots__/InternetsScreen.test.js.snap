// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InternetsScreen should render component 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <WebView
    cacheEnabled={true}
    originWhitelist={
      Array [
        "http://*",
        "https://*",
      ]
    }
    source={
      Object {
        "uri": "https://www.bottlerocketstudios.com/",
      }
    }
    useSharedProcessPool={true}
    useWebKit={true}
  />
</View>
`;
